extensions/amp-ima-video/0.1/test/test-amp-ima-video.js

```diff
@@ -0,0 +1,893 @@
+      const div = document.createElement('div');
+      div.setAttribute('id', 'c');
+      iframe.doc.body.appendChild(div);
+
+      imaVideoObj.imaVideo(iframe.win, {
+        width: 640,
+        height: 360,
+        src: srcUrl,
+        tag: adTagUrl,
+      });
+      const bigPlayDivMock = {
+        style: {
+          display: '',
+        },
+        removeEventListener: function() {},
+      };
+      const removeEventListenerSpy = sandbox.spy(
+          bigPlayDivMock, 'removeEventListener');
+      const adDisplayContainerMock = {initialize: function() {}};
+      const initSpy = sandbox.spy(adDisplayContainerMock, 'initialize');
+      const videoPlayerMock = {load: function() {}};
+      const loadSpy = sandbox.spy(videoPlayerMock, 'load');
+      //const playAdsSpy = sandbox.spy(imaVideoObj, 'playAds');
+      imaVideoObj.setBigPlayDivForTesting(bigPlayDivMock);
+      imaVideoObj.setAdDisplayContainerForTesting(adDisplayContainerMock);
+      imaVideoObj.setVideoPlayerForTesting(videoPlayerMock);
+
+      imaVideoObj.onClick();
+
+      expect(imaVideoObj.getPropertiesForTesting().playbackStarted).to.be.true;
+      expect(imaVideoObj.getPropertiesForTesting().uiTicker)
+          .to.not.be.undefined;
+      expect(removeEventListenerSpy).to.be.calledWith(
+          imaVideoObj.getPropertiesForTesting().interactEvent);
+      expect(bigPlayDivMock.style.display).to.eql('none');
+      expect(initSpy).to.be.called;
+      expect(loadSpy).to.be.called;
+      // TODO - Fix one I figure out how to spy on internals.
+      //expect(playAdsSpy).to.be.called;
+    });
+  });
+
+  it('plays ads with ads manager', () => {
+    return createIframePromise().then(iframe => {
+      const div = document.createElement('div');
+      div.setAttribute('id', 'c');
+      iframe.doc.body.appendChild(div);
+
+      imaVideoObj.imaVideo(iframe.win, {
+        width: 640,
+        height: 360,
+        src: srcUrl,
+        tag: adTagUrl,
+      });
+      //let testPostMessage;
+      // TODO - How do I test messages posted back to the main window?
+      /*window.addEventListener('message', msg => {
+        testPostMessage = msg;
+      });*/
+      window.google = {
+        ima: {
+          ViewMode: {
+            NORMAL: 'normal',
+          },
+        },
+      };
+      const mockAdsManager = {};
+      mockAdsManager.init = function() {};
+      mockAdsManager.start = function() {};
+      const initSpy = sandbox.spy(mockAdsManager, 'init');
+      const startSpy = sandbox.spy(mockAdsManager, 'start');
+      imaVideoObj.setAdsManagerForTesting(mockAdsManager);
+      imaVideoObj.setVideoWidthAndHeightForTesting(100, 200);
+
+      imaVideoObj.playAds();
+
+      expect(initSpy).to.be.calledWith(100, 200, 'normal');
+      // TODO - How do I test messages posted back to the main window? This is
+      // being called before the event is thrown so it fails.
+      //expect(testPostMessage.data).to.have.property('event');
```

Comment:
I don't think you need to test those here. The integration tests in `test-video-players` ensure there events are fired.